### PR TITLE
Fix add-paths lines

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The snsdemo repo was added accidentally.  (Not the whole repo but as a link.)

This looks as if it was caused by an incorrect pattern.

# Changes
- Remove the spurious directory.
- Fix incorrect patterns.

# Tests
We can trigger workflows but given that the workflows do nothing most of the time, the only real test will be to wait for changes that cause non-trivial changes and check whether the bot creates correct outputs.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: .github/workflows/update-proposals.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
           author: gix-bot <gix-bot@users.noreply.github.com>
           branch: bot-proposals-update
           branch-suffix: timestamp
-          add-paths: .github/*/*ml
+          add-paths: rs/proposals/src/canisters/*/api.rs
```
Comment (bitdivine) [inline .github/workflows/update-proposals.yml:39]: This bot should only be generating rust API code for proposal rendering.

Example: None, as this bot hasn't made any PRs yet.

File: .github/workflows/update-snsdemo.yml [hunk 2]
```diff
@@ -64,7 +64,7 @@ jobs:
           author: gix-bot <gix-bot@users.noreply.github.com>
           branch: bot-snsdemo-update
           branch-suffix: timestamp
-          add-paths: .github/*/*ml
+          add-paths: dfx.json
```
Comment (bitdivine) [inline .github/workflows/update-snsdemo.yml:67]: The bot should only be updating the snsdemo version in dfx.json 

Example: https://github.com/dfinity/nns-dapp/pull/3717/files

File: .github/workflows/update-ic.yml [hunk 3]
```diff
@@ -78,6 +78,9 @@ jobs:
         with:
           token: ${{ secrets.GIX_BOT_PAT }}
           base: main
+          add-paths: |
+            dfx.json
+            declarations/*/*.did
           commit-message: Update ic candid files
```
Comment (bitdivine) [inline .github/workflows/update-ic.yml:84]: This should be updating only the IC version in dfx.json and the .did files associated with that commit.

Example: https://github.com/dfinity/nns-dapp/pull/3719/files

**Review by nmattia — APPROVED** [review]
LGTM!

You could nonetheless check out the snsdemo repo in a different direction to be safe and avoid anything being overwritten.

Thought: the comments you've added here on GitHub could be valuable in the code as well

**Comment by bitdivine** [comment]
@nmattia I've started a PR to move the code.  It looks as if to checkout side by side, I also have to change the way the main repo is checked out.  https://github.com/actions/checkout#checkout-multiple-repos-side-by-side so that will be a slightly bigger PR.

**Pull request merged**